### DISCUSSION TITLE
NextJS integration

### DISCUSSION THREAD

**Discussion by amesas** [saleor/saleor#6398] [Q&A] [answered]
¿Are there any plans to give "official" support to NextJS Commerce?

https://nextjs.org/commerce

**Comment by maarcingebala** [comment, marked as answer]
We worked with Next.js on creating an official integration and I think Next.js will release that soon!

**Reply by mirekm** [reply]
@patrys might have a good insight into that effort.. ;)

**Reply by amesas** [reply]
@patrys feel free to share any info :D

Next week we wanna start working on it, and having the hooks available would be interesting!

**Reply by patrys** [reply]
Our initial approach was to build an example showing how Saleor can be paired with Next.js and you can see it here: https://github.com/vercel/next.js/pull/17878

It was closed in favor of two other efforts:
* Switching the Saleor Storefront to Next.js (we're working on this currently)
* Adding Saleor support to the Next.js Commerce project (we're waiting for Commerce to get some bits in place before it's ready to accept alternative backends)

**Reply by saibs** [reply]
by any chance you mean [agnostic UI](https://github.com/vercel/commerce/pull/199) changes in next commerce repo you were waiting for?
So, would there be any integration on the next commerce repo or would [saleor-storefront](https://github.com/mirumee/saleor-storefront) will be the reference implementation?

**Reply by gorjan-mishevski** [reply]
@maarcingebala / @patrys any updates?
Far as I know, Nextjs now accepts PR's for extending the framework commerce API. Bigcommerce and Shopify are done, is Saleor integration on your roadmap by any chance?

**Reply by patrys** [reply]
@gorjan-mishevski we're already working on it! https://github.com/vercel/commerce/pull/277

**Reply by gorjan-mishevski** [reply]
@patrys thank you for the quick update :). Added myself as a subscriber there, you guys are awesome.
